app/controllers/chat_channels_controller.rb

```diff
@@ -76,4 +76,34 @@ def notification_settings
       render_json_error(membership)
     end
   end
+
+  def for_tag
+    params.require(:tag_name)
+
+    tag = Tag.find_by(name: params[:tag_name])
+    raise Discourse::NotFound unless tag
+
+    render_channel_for_chatable(
+      ChatChannel.find_by(chatable: tag)
```

Comment:
I don't have full context but don't we need to specify the `chatable_type` of `"tag"` here?

Reply:
No, it's a polymorphic relationship and it picks up chatable_type automagically